tests/snuba/incidents/test_tasks.py

```diff
@@ -158,3 +160,16 @@ def shutdown_callback(*args, **kwargs):
         assert out.subject == message.subject
         built_message = message.build(self.user.email)
         assert out.body == built_message.body
+
+    def test(self):
+        consumer = QuerySubscriptionConsumer("hi", topic=self.topic)
+        self.run_test(consumer)
+
+    def test_arroyo(self):
+        from sentry.utils.batching_kafka_consumer import create_topics
+
+        kafka_cluster = settings.KAFKA_TOPICS[self.topic]["cluster"]
+        create_topics(kafka_cluster, [self.topic])
```

Comment:
I couldn't get your suggested [create_topic](https://github.com/getsentry/sentry/blob/master/tests/sentry/eventstream/kafka/test_consumer.py#L23-L45) working, it always threw an error on the creation command, but this works fine!

```
________________________________________________________________ HandleSnubaQueryUpdateTest.test_arroyo _________________________________________________________________
tests/snuba/incidents/test_tasks.py:203: in test_arroyo
    with self.create_topic(self.topic):
../../.pyenv/versions/3.8.13/lib/python3.8/contextlib.py:113: in __enter__
    return next(self.gen)
tests/snuba/incidents/test_tasks.py:70: in create_topic
    subprocess.check_call(
../../.pyenv/versions/3.8.13/lib/python3.8/subprocess.py:364: in check_call
    raise CalledProcessError(retcode, cmd)
E   subprocess.CalledProcessError: Command '['docker', 'exec', 'sentry_kafka', 'kafka-topics', '--zookeeper', '127.0.0.1:2181', '--create', '--topic', '4750ba1012d74b729c06585f989ed7d1', '--partitions', '1', '--replication-factor', '1']' returned non-zero exit status 1.
------------------------------------------------------------------------- Captured stderr call --------------------------------------------------------------------------
Exception in thread "main" kafka.zookeeper.ZooKeeperClientTimeoutException: Timed out waiting for connection while in state: CONNECTING
	at kafka.zookeeper.ZooKeeperClient.waitUntilConnected(ZooKeeperClient.scala:271)
	at kafka.zookeeper.ZooKeeperClient.<init>(ZooKeeperClient.scala:125)
	at kafka.zk.KafkaZkClient$.apply(KafkaZkClient.scala:1948)
	at kafka.admin.TopicCommand$ZookeeperTopicService$.apply(TopicCommand.scala:378)
	at kafka.admin.TopicCommand$.main(TopicCommand.scala:56)
	at kafka.admin.TopicCommand.main(TopicCommand.scala)
```

Reply:
Ok. Yeah I'm not a fan of the subprocess.call either, better using the client API to create the topic which is what `create_topics` does. Ideally we would also delete the topic after the test though